dev/tools/localization/gen_l10n.dart

```diff
@@ -289,6 +288,12 @@ bool _isValidClassName(String className) {
   return true;
 }
 
+bool _isDirectoryReadableAndWritable(String statString) {
```

Comment:
This would probably be a little clearer as a function (_isReadableAndWritable) with a FileSystemEntity parameter

Reply:
I made the parameter a FileStat object. That way, the documentation for FileState.modeString() can explain my logic for determining if the String is readable and writable